crud/src/main/java/com/redhat/lightblue/mediator/Mediator.java

```diff
@@ -695,34 +713,47 @@ public BulkResponse bulkRequest(BulkRequest requests) {
             BulkExecutionContext ctx = new BulkExecutionContext(n);
             ctx.setResultSizeThresholds(factory.getMaxResultSetSizeForReadsB(), factory.getWarnResultSetSizeB(), requests);
-
             for (int i = 0; i < n; i++) {
+            	RequestMetrics.Context metricCtx = null;
+            	Response resp = null;
                 Request req = requestList.get(i);
                 if (requests.isOrdered()) {
                     // ordered - only consecutive finds in parallel
                     if (req.getOperation() == CRUDOperation.FIND) {
-                        ctx.futures[i] = executor.submit(getFutureRequest(req));
+                        ctx.futures[i] = executor.submit(getFutureRequest(req, metrics));
                     } else {
                         wait(ctx);
                         switch (req.getOperation()) {
                             case INSERT:
-                                ctx.setResponseAt(i, insert((InsertionRequest) req));
+                            	metricCtx = metrics.startEntityRequest("insert", req.getEntityVersion().getEntity(), req.getEntityVersion().getVersion());
+                            	resp = insert((InsertionRequest) req);
+                                metricCtx.markAllErrorsAndEndRequestMonitoring(resp.getErrors());
+                                ctx.setResponseAt(i, resp);
                                 break;
                             case DELETE:
-                                ctx.setResponseAt(i, delete((DeleteRequest) req));
+                                metricCtx = metrics.startEntityRequest("delete", req.getEntityVersion().getEntity(), req.getEntityVersion().getVersion());
+                                resp = delete((DeleteRequest) req);
+                                metricCtx.markAllErrorsAndEndRequestMonitoring(resp.getErrors());
+                                ctx.setResponseAt(i, resp);
                                 break;
                             case UPDATE:
-                                ctx.setResponseAt(i, update((UpdateRequest) req));
+                                metricCtx = metrics.startEntityRequest("update", req.getEntityVersion().getEntity(), req.getEntityVersion().getVersion());
+                                resp = update((UpdateRequest) req);
+                                metricCtx.markAllErrorsAndEndRequestMonitoring(resp.getErrors());
+                                ctx.setResponseAt(i, resp);
                                 break;
                             case SAVE:
-                                ctx.setResponseAt(i, save((SaveRequest) req));
+                                metricCtx = metrics.startEntityRequest("save", req.getEntityVersion().getEntity(), req.getEntityVersion().getVersion());
+                                resp = save((SaveRequest) req);
+                                metricCtx.markAllErrorsAndEndRequestMonitoring(resp.getErrors());                                
```

Comment:
Similar to future code, this looks good. We could do similar refactoring to reduce duplication would be my only suggestion.